README.md

```diff
@@ -117,8 +117,20 @@ assert(#{ [true]: 1 } === #{ ["true"]: 1 })
 
 assert(#{ [1 + 1]: "two" } === #{ 2: "two" })
 assert(#{ [9 + 1]: "ten" } === #{ ["10"]: "ten" })
+```
+
+`toString()` overrides are invoked.
```

Comment:
also valueOf? In which order? (ie, which hint to ToPrimitive)

Reply:
@ljharb Is that behavior already specified somewhere? If yes, then I can add examples to represent it next to the `toString()` examples.

If not, would the precedence shown below work? (Bear in mind I don't know much about lower-level stuff like `ToPrimitive`.)

```js
function coerceKey(value) {
  if (value == null) {
    throw new Error('seems like this should be an error?')
  }
  switch (typeof value) {
    case 'symbol':
    case 'string':
      return value
    case 'number':
    case 'boolean':
      return String(value)
  }
  // value is a non-null object
  let primValue
  if (Symbol.toPrimitive in value) {
    primValue = value[Symbol.toPrimitive]('string')
  } else if (value.toString !== Object.prototype.toString) {
    primValue = value.toString()
  } else if (value.valueOf !== Object.prototype.valueOf) {
    primValue = value.valueOf()
  } else {
    return String(value)
  }
  switch (typeof primValue) {
    case 'symbol':
    case 'string':
      return primValue
    case 'number':
    case 'boolean':
      return String(primValue)
  }
  throw new Error('seems like this should be an error?')
}
```

If the above precedence seems fine, then I can represent it in the README, with a few examples. Or, maybe we should leave the `toString()` section as is and augment later when the precedence is nailed down?

Reply:
The spec approach here id expect is https://tc39.es/ecma262/#sec-topropertykey - which covers the precedence/ordering.

In JavaScript, one way to simulate it might be `Reflect.ownKeys({ __proto__: null, [x]: true })[0]`.

Reply:
@ljharb The hint in `ToPrimitive` would still be `string`, which would result in `toString` before `valueOf`.

@tiffon If you take a look at [7.1.19 ToPropertyKey](https://tc39.es/ecma262/#sec-topropertykey), it references [7.1.1 ToPrimitive](https://tc39.es/ecma262/#sec-toprimitive) with hint `String`, which results in a `string` hint to [7.1.1.1 OrdinaryToPrimitive](https://tc39.es/ecma262/#sec-ordinarytoprimitive).

This means that when converting a value to a property key (via `ToPropertyKey`), `toString` will be preferred over `valueOf` when converting the value (like your code example also mentions). Do you mind adding some additional examples that demonstrate the precedence of `@@toPrimitive`, `toString`, and `valueOf`?